File: top-interview-150/0169-majority-element/solution.py
# 169. Majority Element
# https://leetcode.com/problems/majority-element/
# Accepted: 2026-07-18T20:06:09.000Z
# Language: Python3
# Collection: top-interview-150
# Runtime: 7 ms · Beats 57.3%
# Memory: 21.1 MB · Beats 49.39%
# Submission: https://leetcode.com/submissions/detail/2072770273/

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        majority = nums[0]
        compteur = 1

        for i in range(1,len(nums)) : 

            current = nums[i]
            if current == majority : 
                compteur +=1
            else : 
                compteur -=1
            
            if compteur < 0 : 
                compteur = -compteur
                majority = current
            
        return majority

File: top-interview-150/0169-majority-element/solution.cpp
// 169. Majority Element
// https://leetcode.com/problems/majority-element/
// Accepted: 2026-09-24T22:49:33.000Z
// Language: C++
// Collection: top-interview-150
// Runtime: 0 ms · Beats 100%
// Memory: 42.1 MB · Beats 7.09%
// Submission: https://leetcode.com/submissions/detail/2152471370/

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int counter = 1 ;
        int majority = nums[0] ; 
        for (int i = 1 ; i< nums.size(); i++){
            if (nums[i] == majority){
                counter +=1 ; 
            } else {
                counter -=1 ;
            }
            if (counter < 0){
                counter = 1;
                majority = nums[i];
            }
        }
        return majority ; 
    }
};
